c_stubs/xenctrlext_stubs.c

```diff
@@ -195,6 +195,17 @@ CAMLprim value stub_xenctrlext_domain_set_target(value xch,
 	CAMLreturn(Val_unit);
 }
 
+CAMLprim value stub_xenctrlext_physdev_map_pirq(value xch,
+                     value domid,
+                     value irq)
+{
+    CAMLparam3(xch, domid, irq);
+    int pirq = Int_val(irq);
+    int retval = xc_physdev_map_pirq(_H(xch), _D(domid), irq, &pirq);
```

Comment:
If this can take "long" to execute, then we should wrap it in a blocking section, as in `stub_xenctrlext_get_max_nr_cpus` above.

Reply:
Under the PCI specification, it can take up to 90s to complete.  (This is of course ridiculous, and Xen will give up far earlier than that, but that is what the spec requires...)

Reply:
In that case, I will table an amendment to add the blocking section.